### ISSUE TITLE
Add ability to automatically display import paths in the interface

### ISSUE THREAD

**Issue by rey-wright** [storybookjs/storybook#7772]
<img width="294" alt="Screen Shot 2019-08-14 at 12 28 52 PM" src="https://user-images.githubusercontent.com/52427513/63039798-134fa480-be92-11e9-9b63-7cf46cc35c4b.png">
https://react-styleguidist.js.org/examples/basic/

Above screenshot (link) is from React Styleguidist. Styleguidist automatically displays a copy / paste link of the import path for end user convenience (developers do not need to assist this on a per component basis, this is just configured somewhere once and every component gets this copy & paste feature). I've also provided a link above where you can see how this works in action.

@shilman

**Comment by shilman** [comment]
Love this @rey-wright 😍 Hopefully we can get that for you as soon as the 5.2 release is out the door!

**Comment by kevinSuttle** [comment]
Huh. That was easy. 

<img width="905" alt="Image 2019-08-14 at 10 01 34 PM" src="https://user-images.githubusercontent.com/95672/63067894-1c1a9780-bedf-11e9-85af-dbb6b46095dd.png">


https://react-styleguidist.js.org/docs/configuration.html#getcomponentpathline

**Comment by rey-wright** [comment]
Is there a timeline for when this feature would come out? I had heard it was February after 5.3 was released. I only ask as we want to make a component for internal usage that does this (albeit by passing a string in the mdx file we use it in as the path), so before I work on that, wanted to just see if I should work on that or not.

**Comment by shilman** [comment]
Your best bet is to build it yourself or contribute it. 6.0 is slated for late spring, so that's the earliest stable release it could go out with.

**Comment by folz** [comment]
I have a need for this feature and would be down to build it out myself. I'm unfamiliar with storybook's source code. @shilman, could you point me to where you'd recommend starting (or someone else who has context)?

**Comment by shilman** [comment]
@folz I'd be happy to. Join in in the Storybook Discord? https://discord.gg/UUt2PJb

**Comment by soullivaneuh** [comment]
Some suggestions:

First, it would be great to see the import on the related code block directly:

![image](https://user-images.githubusercontent.com/1698357/111783153-34704680-88ba-11eb-9791-630e9b80f476.png)

Thanks to that, the copy past may be easier.

Second, it would be great to detect or configure the way we want our users to import the components.
For example, components of library are most of the time exported trough an index file. The example should looks like:

```typescript
import { MyComponent } from '@company/components';
```

What do you think?

**Comment by reywright** [comment]
@soullivaneuh There's no longer a dire need for Storybook to support this as you can extend docs to support it.

https://github.com/cbinsights/form-design-system/blob/master/.storybook/preview.js#L23

Notice here that I'm extending the base level docs page to include this custom "ImportPath" component, which has access to the story information through context. The only possible thing they could add here would be the actual path to the component / file you're documenting but this was sufficient enough for my use case.

https://cbinsights.github.io/form-design-system/fds-components/?path=/docs/components-button--primary

Here's our design system you can look at. You'll notice all of the component we document have this import path now, it's specifically because of this `ImportPath` component we're specifying only here.

**Comment by sarahbethfederman** [comment]
I mean, I would like to see this included in the future, hopefully with a bit more logic around default exports, subcomponents, and multiple entrypoints. I am not sure if it's fair to say that just being able to do `import { ${componentName} } from '@mylib/components';` (https://github.com/cbinsights/form-design-system/blob/master/src/util/storybook-docs/ImportPath.jsx#L14) on every page really solves this problem

**Comment by reywright** [comment]
@sarahbethfederman Yeah totally that's fair.

* When I created this ticket it was for providing a simple UI as well as a very simplistic implementation.
* I would almost rather create a separate ticket just to request merely include this data in the context now (pathname of the component, subcomponents, etc) as a first step, so it becomes less of a question of how this should look UI wise, to be a task at a later time. I'm assuming that's part of the reason this hasn't been picked up, is that it's too broad of an ask.

**Comment by shilman** [comment]
If it's cool with you @reywright let's keep it all in one place. When somebody gets around to implementing this, it will be useful to see all the context together.

**Comment by ricovitch** [comment]
Hi I'm also interested in a standard storybook way around documenting imports.
I went with a solution inspired from @rey-wright `ImportPath` component, but by providing a `components` prop like for the `ArgsTable` one, and using storybook docs `Source` block instead of custom css/js.
Here is my version of the `ImportPath` component
```js
import React from "react";
import { Source } from "@storybook/addon-docs/blocks";

export const ImportPath = ({ components }) => {
  const getComponents = () => Object.keys(components).join(", ");
  const path = `import { ${getComponents()} } from "@airbus/components-react"`;

  return <Source language="js" code={path}></Source>;
};

export default ImportPath;
```

**Comment by reywright** [comment]
That's sweet @ricovitch I might have to change my component up a bit! Is there any particular reason you're not using context to infer the component? I'm assuming it's because you have multiple components?

**Comment by ricovitch** [comment]
> That's sweet @ricovitch I might have to change my component up a bit! Is there any particular reason you're not using context to infer the component? I'm assuming it's because you have multiple components?

Yes mainly for being able to pass a list of components. I tried hard to find a way to do that with `DocsContext` without success. So I went with something simple, and consistent with the `ArgsTable` docs block.

**Comment by markrocks** [comment]
> https://github.com/cbinsights/form-design-system/blob/master/src/util/storybook-docs/Layout.jsx#L20

I would love to get more context on where this solution is implemented. Unfortunately the original references this was based on by @ray-wright do not seem to be available. Can any on provide me a quick overview on where I can implement this ImportPath based solution?

**Comment by reywright** [comment]
@markrocks https://github.com/cbinsights/form-design-system/blob/695e43d5b0eb0085b9e443c7055d63ce2d05f53c/src/util/storybook-docs/ImportPath.tsx

It's imperfect but it's worked decently for us so far. Let me know if you have questions.

**Comment by krutoo** [comment]
Hi to all, in my case i export from library miltiple components (as example Button, Modal and other)
I want to see in docs page import that depends on compoenent entry file placement like:

```js
import { FooBar } from '@some-lib/foo-bar'
```

This solution looks pretty:

> @markrocks https://github.com/cbinsights/form-design-system/blob/695e43d5b0eb0085b9e443c7055d63ce2d05f53c/src/util/storybook-docs/ImportPath.tsx

...but i don't know how to get string with story absolute file path

How can i get it?
